### PR TITLE
fix(view): turn off view encapsulation

### PR DESCRIPTION
**Pull request by mindreeper2420**
set `ViewEncapsulation:None` in the app and each component in order to eliminate bloated styles and improve the ability to extend off of PatternFly styles. Reorganized shared styles in order to use proper directory structure, set minification of compiled CSS for all instances and cleaned out duplicate shared styles.

As this PR is **massive**, please have more than 2 approve before merging - for safety :-).

### CODE REVIEW

**Comment by fabric8cd** [comment]
@mindreeper2420 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by mindreeper2420** [comment]
I tested the snapshot and noticed that some styles were off. Once I reloaded, everything was as it should be. Due to the relocation of files and changing of how things are built, any cached CSS may cause an issue on first load.

**Comment by joshuawilson** [comment]
Looks like the main OSIO icon on the Account Home screen is missing.

**Comment by fabric8cd** [comment]
@mindreeper2420 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by SMahil** [comment]
screen shot 2017-07-07 at 11 13 15 am

Left and right is not proper because if height: 100vh; height:100% solves it

**Comment by mindreeper2420** [comment]
@SMahil I'm not seeing that issue on my local build using multiple browsers with caching off. Can you check the Analyze page as well? That is `100vh;` in prod right now, without issue.

**Comment by mindreeper2420** [comment]
@joshuawilson locally, it shows up for me. I ran `npm run build:prod` and found that the hashes introduced on production builds were causing the files to not be recognizable. I've updated and pushed.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by mindreeper2420** [comment]
@joshuawilson - turns out, the real problem was a change that happened with the branding(?) update. There was a change in the typescript file that had it looking for the wrong path. I didn't know that had made it to prod, so I didn't look for it at first.

**Comment by mindreeper2420** [comment]
@joshuawilson that also doesn't seem to have any affect on dev environments - not sure how it was set up, but images paths for branding are something to look out for moving forward.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-8 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by SMahil** [comment]
@mindreeper2420 I checked on Analyze page too and its still there. I checked in incognito mode on different browsers

**Comment by mindreeper2420** [comment]
@SMahil are you viewing in Firefox? I ran it there and noticed the issue. `height: 100vh;` works on Safari and Chrome, but not Firefox, and `height: 100%;` doesn't work on Safari or Chrome, but does on Firefox :-)

**Comment by SMahil** [comment]
@mindreeper2420 yes i checked on firefox too

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-9 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by joshuawilson** [comment]
The main OSIO image is gone again when I use the link above.

File: config/webpack.common.js [hunk 1]
```diff
@@ -189,7 +189,7 @@ module.exports = function (options) {
             }, {
               loader: 'css-loader',
               options: {
-                minimize: isProd,
+                minimize: true,
```
Comment (joshuawilson) [inline config/webpack.common.js]: The point of having a variable here is so that during development you can have the styles un-minimized.

File: README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 You need to setup your shell to point to the right cluster so that it can talk to the required back end services like KeyCloak, WIT, Forge, OpenShift etc.
 
-We provide various sample environments out of the box which make it easier to get started. They are all located as bash scripts in `environments`. 
+We provide various sample environments out of the box which make it easier to get started. They are all located as bash scripts in `environments`.
```
Comment (joshuawilson) [inline README.md]: This should all be in a different commit and different PR.

**Comment by joshuawilson** [comment]
There are 121 files changes and a lot of things that are not a part of the title.

File: config/webpack.common.js (hunk 1 above)
Comment (mindreeper2420) [inline config/webpack.common.js]: it doesn't make a difference though in terms of inspecting the code - the inspectors pull everything up as they should. The point for doing this is so that the dev environment can mimic production - I found a few bugs by changing this to always minimized.

File: README.md (hunk 2 above)
Comment (mindreeper2420) [inline README.md]: I'll revert it back to what is was.

**Comment by mindreeper2420** [comment]
The OSIO image should be fixed as a separate PR. No changes were made in regards to that as part of this work, so it is an issue in Master.

**Comment by mindreeper2420** [comment]
@SMahil I've set the heights to be `100%` for both the `.home-wrapper` and `.analyze-overview-wrapper` classes. The `<body>` does not have a color applied, so anything that extends beyond those wrappers will be the standard white.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-10 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-11 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-12 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-13 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

File: src/app/about-modal/about-modal.component.less [hunk 3]
```diff
@@ -1,6 +1,6 @@
-@import (reference) '../../assets/stylesheets/_base';
-.modal-dialog-f8 { width: 696px; }
-.about-modal-f8 {
+@import (reference) '../../assets/stylesheets/shared/osio.less';
+.io-modal-dialog { width: 696px; }
```
Comment (joshuawilson) [inline src/app/about-modal/about-modal.component.less]: Why did you change this? The namespace we are using is `f8`. `io` is too generic.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-14 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

File: src/app/about-modal/about-modal.component.less (hunk 3 above)
Comment (mindreeper2420) [inline src/app/about-modal/about-modal.component.less]: The other `io` specific variables all use `io-` currently. This was change was made to match those. If all of the classes should be prefixed with `f8`, I can do that separately.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-15 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-16 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by mindreeper2420** [comment]
@joshuawilson updated with the latest changes from Master, so now only 72 files have been changed.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-17 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-18 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io

File: src/app/profile/resources/resources.component.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<div id="resources" class="container content">
```
Comment (joshuawilson) [inline src/app/profile/resources/resources.component.html]: I think you may have accidentally added this files back in. The Resource component was deleted.

**Comment by mindreeper2420** [comment]
@joshuawilson yes, one of the rebases appears to have added it back. I removed the component again and updated the PR.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-1645-19 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1645-fabric8-ui.badger.fabric8.io
